pkg/internal/server/grpc/common_test.go

```diff
@@ -45,64 +47,70 @@ import (
 type testHarness struct {
 	*runtime.TestHarness
 
-	SecurityToken string
-	DriverName    string
-	Audience      string
-	Namespace     string
-
-	fakeKubeClient     *fake.Clientset
-	fakeSnapshotClient *fakesnapshot.Clientset
-	fakeCBTClient      *fakecbt.Clientset
-
-	grpcServer *grpc.Server
-	listener   *bufconn.Listener
+	Audience                string
+	DriverName              string
+	Namespace               string
+	SecretName              string
+	SecretNs                string
+	SecurityToken           string
+	VolumeSnapshotClassName string
+
+	FakeKubeClient     *fake.Clientset
+	FakeSnapshotClient *fakesnapshot.Clientset
+	FakeCBTClient      *fakecbt.Clientset
+
+	lenFakeKubeClientReactionChain     int
+	lenFakeSnapshotClientReactionChain int
+	lenFakeCBTClientReactionChain      int
+
+	mockCSIDriverConn *grpc.ClientConn
+	grpcServer        *grpc.Server
+	listener          *bufconn.Listener
 }
 
 func newTestHarness() *testHarness {
 	return &testHarness{
-		SecurityToken: "securityToken",
-		DriverName:    "driver",
-		Audience:      "audience",
-		Namespace:     "namespace",
+		Audience:                "audience",
+		DriverName:              "driver",
+		Namespace:               "namespace",
+		SecretName:              "secret-name",
+		SecretNs:                "secret-ns",
+		SecurityToken:           "securityToken",
+		VolumeSnapshotClassName: "csi-snapshot-class",
 	}
 }
 
 func (th *testHarness) WithMockCSIDriver(t *testing.T) *testHarness {
 	th.TestHarness = runtime.NewTestHarness().WithMockCSIDriver(t)
+	th.mockCSIDriverConn = th.MockCSIDriverConn
 	return th
 }
 
 func (th *testHarness) WithFakeClientAPIs() *testHarness {
-	th.fakeKubeClient = th.makeFakeKubeClient()
-	th.fakeSnapshotClient = th.makeFakeSnapshotClient()
-	th.fakeCBTClient = th.makeFakeCBTClient()
+	th.FakeKubeClient = th.makeFakeKubeClient()
+	th.FakeSnapshotClient = th.makeFakeSnapshotClient()
+	th.FakeCBTClient = th.makeFakeCBTClient()
+
+	th.FakeCountReactors() // establish the baseline
+
 	return th
 }
 
 func (th *testHarness) Runtime() *runtime.Runtime {
 	return &runtime.Runtime{
-		CBTClient:      th.fakeCBTClient,
-		KubeClient:     th.fakeKubeClient,
-		SnapshotClient: th.fakeSnapshotClient,
+		CBTClient:      th.FakeCBTClient,
+		KubeClient:     th.FakeKubeClient,
+		SnapshotClient: th.FakeSnapshotClient,
 		DriverName:     th.DriverName,
+		CSIConn:        th.mockCSIDriverConn,
 	}
 }
 
-func (th *testHarness) RuntimeWithMockCSIDriver(t *testing.T) *runtime.Runtime {
-	assert.NotNil(t, th.MockCSIDriverConn, "needs WithMockCSIDriver")
-	rt := th.Runtime()
-	rt.CSIConn = th.MockCSIDriverConn
-	rt.Args = th.RuntimeArgs()
-	return rt
-}
```

Comment:
`WithMockCSIDriver` now sets `th.mockCSIDriverConn` used in `th.Runtime` so this is not needed any more.